pkg/controller/certificates/readiness/readiness_controller.go

```diff
@@ -163,9 +163,17 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		crt.Status.NotBefore = &notBefore
 		crt.Status.NotAfter = &notAfter
 		// calculate how long before the certificate expiry time the certificate
-		// should be renewed
-		renewBefore := certificates.RenewBeforeExpiryDuration(crt.Status.NotBefore.Time, crt.Status.NotAfter.Time, crt.Spec.RenewBefore, c.defaultRenewBeforeExpiryDuration)
-		renewalTime := metav1.NewTime(notAfter.Add(-1 * renewBefore))
+		// should be renewed.
+		// TODO: This default should be applied by one of the API defaulting
+		// mechanisms: OpenAPI defaulting or defaulting webhook. See:
+		// https://github.com/jetstack/cert-manager/pull/3466
+		renewBefore := c.defaultRenewBeforeExpiryDuration
+		if crt.Spec.RenewBefore != nil {
+			renewBefore = crt.Spec.RenewBefore.Duration
+		}
+		renewalTime := metav1.NewTime(
+			certificates.CalculateRenewalTime(crt.Status.NotBefore.Time, crt.Status.NotAfter.Time, renewBefore),
+		)
```

Comment:
If I recall correctly, the `crt.Status.RenewalTime` field is purely observational and not used to drive the behaviour of the `trigger` controller

Reply:
Agreed: we might want to follow  @irbekrm's example in https://github.com/jetstack/cert-manager/pull/3665/files#r594555205, where she removed the reliance on the status observed state (`RenewalTime`). Instead, she parses the X.509 cert and uses the notBefore and notAfter fields in order to recompute the renewalTime.